src/Components/Blueprints/BlueprintsSideBar.tsx

```diff
@@ -22,72 +23,132 @@ type blueprintProps = {
   blueprints: BlueprintItem[] | undefined;
   selectedBlueprint: string | undefined;
   setSelectedBlueprint: Dispatch<SetStateAction<string | undefined>>;
+  filter: string | undefined;
+  setFilter: Dispatch<SetStateAction<string>>;
+};
+
+type blueprintSearchProps = {
+  filter: string | undefined;
+  setFilter: Dispatch<SetStateAction<string>>;
+  blueprintsCount: number;
+};
+
+type emptyBlueprintStateProps = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon: React.ComponentClass<SVGIconProps, any>;
+  action: React.ReactNode;
+  titleText: string;
+  bodyText: string;
 };
 
 const BlueprintsSidebar = ({
   blueprints,
   selectedBlueprint,
   setSelectedBlueprint,
+  filter,
+  setFilter,
 }: blueprintProps) => {
-  const [blueprintFilter, setBlueprintFilter] = useState('');
+  const blueprintsCount = blueprints?.length || 0;
```

Comment:
This is just the fetched data length, we can use the meta object for the query search count :)

Reply:
You're completely right! :)
Adding that.